server/src/test/java/org/opensearch/gateway/ClusterStateUpdatersTests.java

```diff
@@ -269,6 +275,214 @@ public void testUpdateRoutingTable() {
         }
     }
 
+    public void testSkipRoutingTableUpdateWhenRemoteRecovery() {
```

Comment:
can you add some comments in the test with respect to different scenarios being tested.

Reply:
sure, done